### PR TITLE
feat(dpp): all changes from v24-wasm-integration

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

js components not functioning properly 

## What was done?

Lots of things

## How Has This Been Tested?

Run tests, run test suite from another branch

## Breaking Changes

Some, probably

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs [hunk 1]
```diff
@@ -57,8 +57,8 @@ pub struct DataContractUpdateTransition {
     pub protocol_version: u32,
     #[serde(rename = "type")]
     pub transition_type: StateTransitionType,
-    // we want to skip serialization of transitions, as we does it manually in `to_object()`  and `to_json()`
-    #[serde(skip_serializing)]
+    // we want to skip serialization of transitions, as we do it manually in `to_object()`  and `to_json()`
+    // #[serde(skip_serializing)]
```
Comment (shumkov) [inline packages/rs-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs:61]: Just comment?

File: packages/rs-dpp/src/data_trigger/reward_share_data_triggers/mod.rs [hunk 2]
```diff
@@ -32,38 +31,50 @@ where
...
-                == context.owner_id.to_buffer()
-        });
-
-        if !owner_id_in_sml {
+        let is_valid_master_node = context
+            .state_repository
+            .is_in_the_valid_master_nodes_list(context.owner_id.to_buffer())
+            .await?;
+
+        // // Do not allow creating document if ownerId is not in SML
```
Comment (shumkov) [inline packages/rs-dpp/src/data_trigger/reward_share_data_triggers/mod.rs:64]: We should add a todo here

File: packages/rs-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs (hunk 1 above)
Comment (antouhou) [reply packages/rs-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs:61]: I'm not sure what to do about this one - this one was done for a reason, but it broke stuff. So while we're not 100% sure (it seems like it doesn't) it doesn't break anything I would keep it here

Comment (shumkov) [reply packages/rs-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs:61]: Let's bring it back then. My point is that we shouldn't commit commented code.

**Comment by shumkov** [comment]
@antouhou what we are going to do with this PR?

**Comment by antouhou** [comment]
Closing, as it has been split into several smaller PRs

**Comment by shumkov** [comment]
It was merge with multiple separate PRs

**Pull request closed without merging**